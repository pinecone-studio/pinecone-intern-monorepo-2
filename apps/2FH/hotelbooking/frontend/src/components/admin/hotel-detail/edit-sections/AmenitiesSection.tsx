'use client';
import { Label } from '@/components/ui/Label';
import { Checkbox } from '@/components/ui/Checkbox';

interface AmenitiesSectionProps {
  formData: any;
  handleInputChange: (_field: string, _value: any) => void;
}

export const AmenitiesSection = ({ formData, handleInputChange }: AmenitiesSectionProps) => {
  const availableAmenities = [
    'POOL',
    'GYM',
    'RESTAURANT',
    'BAR',
    'WIFI',
    'PARKING',
    'FITNESS_CENTER',
    'BUSINESS_CENTER',
    'MEETING_ROOMS',
    'CONFERENCE_ROOMS',
    'ROOM_SERVICE',
    'AIR_CONDITIONING',
    'AIRPORT_TRANSFER',
    'FREE_WIFI',
    'FREE_PARKING',
    'FREE_CANCELLATION',
    'SPA',
    'PETS_ALLOWED',
    'SMOKING_ALLOWED',
    'LAUNDRY_FACILITIES',
  ];

  return (
    <div className="space-y-4">
      <Label>Amenities</Label>
      <div className="grid grid-cols-2 gap-3 max-h-60 overflow-y-auto">
        {availableAmenities.map((amenity) => (
          <div key={amenity} className="flex items-center space-x-2">
            <Checkbox
              id={amenity}
              checked={formData.amenities.includes(amenity)}
              onCheckedChange={(checked) => {
                if (checked) {
                  handleInputChange('amenities', [...formData.amenities, amenity]);
                } else {
                  handleInputChange(
                    'amenities',
                    formData.amenities.filter((a: string) => a !== amenity)
                  );
                }
              }}
            />
            <Label htmlFor={amenity} className="text-sm">
              {amenity.replace(/_/g, ' ')}
            </Label>
          </div>
        ))}
      </div>
    </div>
  );
};
