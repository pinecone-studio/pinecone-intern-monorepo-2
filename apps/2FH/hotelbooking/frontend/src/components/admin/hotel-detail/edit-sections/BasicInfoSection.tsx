'use client';
import { Input } from '@/components/ui/Input';
import { Textarea } from '@/components/ui/Textarea';
import { Label } from '@/components/ui/Label';

interface BasicInfoSectionProps {
  formData: any;
  handleInputChange: (_field: string, _value: any) => void;
}

export const BasicInfoSection = ({ formData, handleInputChange }: BasicInfoSectionProps) => {
  return (
    <div className="space-y-4">
      <div>
        <Label htmlFor="name" className="text-sm font-medium text-gray-700">
          Hotel Name
        </Label>
        <Input id="name" value={formData.name} onChange={(e) => handleInputChange('name', e.target.value)} />
      </div>
      <div>
        <Label htmlFor="description" className="text-sm font-medium text-gray-700">
          Description
        </Label>
        <Textarea id="description" value={formData.description} onChange={(e) => handleInputChange('description', e.target.value)} rows={4} />
      </div>
      <div className="grid grid-cols-2 gap-4">
        <div>
          <Label htmlFor="phone" className="text-sm font-medium text-gray-700">
            Phone Number
          </Label>
          <Input id="phone" value={formData.phone} onChange={(e) => handleInputChange('phone', e.target.value)} />
        </div>
        <div>
          <Label htmlFor="stars" className="text-sm font-medium text-gray-700">
            Stars (Read-only)
          </Label>
          <Input id="stars" type="number" min="1" max="5" value={formData.stars} disabled className="bg-gray-100" />
        </div>
      </div>
      <div className="grid grid-cols-2 gap-4">
        <div>
          <Label htmlFor="country" className="text-sm font-medium text-gray-700">
            Country
          </Label>
          <Input id="country" value={formData.country} onChange={(e) => handleInputChange('country', e.target.value)} placeholder="Enter country..." />
        </div>
        <div>
          <Label htmlFor="city" className="text-sm font-medium text-gray-700">
            City
          </Label>
          <Input id="city" value={formData.city} onChange={(e) => handleInputChange('city', e.target.value)} placeholder="Enter city name..." />
        </div>
      </div>
      <div>
        <Label htmlFor="location" className="text-sm font-medium text-gray-700">
          Location
        </Label>
        <Input id="location" value={formData.location} onChange={(e) => handleInputChange('location', e.target.value)} />
      </div>
      <div>
        <Label htmlFor="rating" className="text-sm font-medium text-gray-700">
          User Rating (Read-only)
        </Label>
        <Input id="rating" type="number" step="0.1" min="0" max="10" value={formData.rating} disabled className="bg-gray-100" />
      </div>
    </div>
  );
};
