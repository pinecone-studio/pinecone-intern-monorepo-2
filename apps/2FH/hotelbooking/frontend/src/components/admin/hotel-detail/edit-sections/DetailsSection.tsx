'use client';
import { Button } from '@/components/ui/Button';
import { Input } from '@/components/ui/Input';
import { Textarea } from '@/components/ui/Textarea';
import { Label } from '@/components/ui/Label';

interface DetailsSectionProps {
  formData: any;
  handleInputChange: (_field: string, _value: any) => void;
}

export const DetailsSection = ({ formData, handleInputChange }: DetailsSectionProps) => {
  return (
    <div className="space-y-6">
      <div>
        <Label className="text-lg font-medium">About</Label>
        <Textarea value={formData.description} onChange={(e) => handleInputChange('description', e.target.value)} rows={4} placeholder="Enter hotel description..." />
      </div>

      <div>
        <Label className="text-lg font-medium">What You Need to Know</Label>
        <div className="space-y-3">
          {formData.optionalExtras &&
            formData.optionalExtras.map((extra: any, index: number) => (
              <div key={index} className="border p-4 rounded-lg">
                <div className="flex justify-between items-center mb-3">
                  <h4 className="font-medium">Item {index + 1}</h4>
                  <Button
                    type="button"
                    variant="outline"
                    size="sm"
                    onClick={() => {
                      const newExtras = formData.optionalExtras.filter((_: any, i: number) => i !== index);
                      handleInputChange('optionalExtras', newExtras);
                    }}
                    className="text-red-600 hover:text-red-700"
                  >
                    Remove
                  </Button>
                </div>
                <div className="space-y-3">
                  <div>
                    <Label>What You Need to Know</Label>
                    <Input
                      value={extra.youNeedToKnow || ''}
                      onChange={(e) => {
                        const newExtras = [...formData.optionalExtras];
                        newExtras[index] = { ...extra, youNeedToKnow: e.target.value };
                        handleInputChange('optionalExtras', newExtras);
                      }}
                      placeholder="Enter what guests need to know..."
                    />
                  </div>
                  <div>
                    <Label>Additional Information</Label>
                    <Textarea
                      value={extra.weShouldMention || ''}
                      onChange={(e) => {
                        const newExtras = [...formData.optionalExtras];
                        newExtras[index] = { ...extra, weShouldMention: e.target.value };
                        handleInputChange('optionalExtras', newExtras);
                      }}
                      rows={2}
                      placeholder="Enter additional information..."
                    />
                  </div>
                </div>
              </div>
            ))}
          <Button
            type="button"
            variant="outline"
            onClick={() => {
              const newExtras = formData.optionalExtras || [];
              handleInputChange('optionalExtras', [...newExtras, { youNeedToKnow: '', weShouldMention: '' }]);
            }}
            className="w-full"
          >
            Add New Item
          </Button>
        </div>
      </div>

      <div>
        <Label className="text-lg font-medium">Languages Spoken</Label>
        <div className="space-y-2">
          {formData.languages.map((language: string, index: number) => (
            <div key={index} className="flex gap-2">
              <Input
                value={language}
                onChange={(e) => {
                  const newLanguages = [...formData.languages];
                  newLanguages[index] = e.target.value;
                  handleInputChange('languages', newLanguages);
                }}
                placeholder="Enter language..."
              />
              <Button
                type="button"
                variant="outline"
                size="sm"
                onClick={() => {
                  handleInputChange(
                    'languages',
                    formData.languages.filter((_: string, i: number) => i !== index)
                  );
                }}
              >
                Remove
              </Button>
            </div>
          ))}
          <Button
            type="button"
            variant="outline"
            onClick={() => {
              handleInputChange('languages', [...formData.languages, '']);
            }}
          >
            Add Language
          </Button>
        </div>
      </div>
    </div>
  );
};
