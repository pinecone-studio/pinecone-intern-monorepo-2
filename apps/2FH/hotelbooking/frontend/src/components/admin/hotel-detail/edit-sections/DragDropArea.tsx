import React from 'react';
import { Button } from '@/components/ui/button';

interface DragDropAreaProps {
  dragActive: boolean;
  uploading: boolean;
  fileInputRef: React.RefObject<HTMLInputElement>;
  handleDragEnter: (_e: React.DragEvent) => void;
  handleDragLeave: (_e: React.DragEvent) => void;
  handleDragOver: (_e: React.DragEvent) => void;
  handleDrop: (_e: React.DragEvent) => void;
  handleFileInput: (_e: React.ChangeEvent<HTMLInputElement>) => void;
}

export const DragDropArea = ({ dragActive, uploading, fileInputRef, handleDragEnter, handleDragLeave, handleDragOver, handleDrop, handleFileInput }: DragDropAreaProps) => (
  <div
    className={`border-2 border-dashed rounded-lg p-8 text-center transition-colors ${dragActive ? 'border-blue-500 bg-blue-50' : 'border-gray-300 hover:border-gray-400'}`}
    onDragEnter={handleDragEnter}
    onDragLeave={handleDragLeave}
    onDragOver={handleDragOver}
    onDrop={handleDrop}
  >
    <div className="space-y-4">
      <div className="mx-auto w-16 h-16 bg-gray-100 rounded-full flex items-center justify-center">
        <svg className="w-8 h-8 text-gray-400" fill="none" stroke="currentColor" viewBox="0 0 24 24">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
        </svg>
      </div>

      <div>
        <p className="text-lg font-medium text-gray-900">{uploading ? 'Uploading...' : 'Drop images here or click to upload'}</p>
        <p className="text-sm text-gray-500 mt-1">Supports JPG, PNG, GIF up to 10MB each</p>
      </div>

      <Button type="button" variant="outline" onClick={() => fileInputRef.current?.click()} disabled={uploading} className="mt-4">
        {uploading ? 'Uploading...' : 'Choose Files'}
      </Button>

      <input ref={fileInputRef} type="file" multiple accept="image/*" onChange={handleFileInput} className="hidden" />
    </div>
  </div>
);
