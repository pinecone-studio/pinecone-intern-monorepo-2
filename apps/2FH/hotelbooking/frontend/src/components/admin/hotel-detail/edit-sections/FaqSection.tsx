'use client';
import { Button } from '@/components/ui/button';
import { Input } from '@/components/ui/input';
import { Textarea } from '@/components/ui/Textarea';
import { Label } from '@/components/ui/Label';

interface FAQSectionProps {
  formData: any;
  handleInputChange: (_field: string, _value: any) => void;
}

export const FAQSection = ({ formData, handleInputChange }: FAQSectionProps) => {
  return (
    <div className="space-y-4">
      {formData.faq.map((faq: any, index: number) => (
        <div key={index} className="border p-4 rounded-lg">
          <div className="flex justify-between items-center mb-3">
            <h4 className="font-medium">FAQ {index + 1}</h4>
            <Button
              type="button"
              variant="outline"
              size="sm"
              onClick={() => {
                handleInputChange(
                  'faq',
                  formData.faq.filter((_: any, i: number) => i !== index)
                );
              }}
              className="text-red-600 hover:text-red-700"
            >
              Remove
            </Button>
          </div>
          <div className="space-y-3">
            <div>
              <Label>Question</Label>
              <Input
                value={faq.question}
                onChange={(e) => {
                  const newFAQ = [...formData.faq];
                  newFAQ[index] = { ...faq, question: e.target.value };
                  handleInputChange('faq', newFAQ);
                }}
              />
            </div>
            <div>
              <Label>Answer</Label>
              <Textarea
                value={faq.answer}
                onChange={(e) => {
                  const newFAQ = [...formData.faq];
                  newFAQ[index] = { ...faq, answer: e.target.value };
                  handleInputChange('faq', newFAQ);
                }}
                rows={3}
              />
            </div>
          </div>
        </div>
      ))}
      <Button
        type="button"
        variant="outline"
        onClick={() => {
          handleInputChange('faq', [...formData.faq, { question: '', answer: '' }]);
        }}
        className="w-full"
      >
        Add New FAQ
      </Button>
    </div>
  );
};
