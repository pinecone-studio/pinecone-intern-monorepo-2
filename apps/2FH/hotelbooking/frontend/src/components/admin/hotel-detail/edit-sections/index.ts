export { BasicInfoSection } from './BasicInfoSection';
export { LocationSection } from './LocationSection';
export { AmenitiesSection } from './AmenitiesSection';
export { LanguagesSection } from './LanguagesSection';
export { PoliciesSection } from './PoliciesSection';
export { FAQSection } from './FaqSection';
export { DetailsSection } from './DetailsSection';
export { ImagesSection } from './ImagesSection';
