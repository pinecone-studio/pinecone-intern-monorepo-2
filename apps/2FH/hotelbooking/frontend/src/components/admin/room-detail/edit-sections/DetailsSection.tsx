/* eslint-disable  */ 
'use client';
import { Label } from '@/components/ui/label';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';

interface DetailsSectionProps {
  room: any;
  handleInputChange: (field: string, value: any) => void;
}

export const DetailsSection = ({ room, handleInputChange }: DetailsSectionProps) => {
  const statusOptions = [
    { value: 'cancelled', label: 'Cancelled' },
    { value: 'booked', label: 'Booked' },
    { value: 'pending', label: 'Pending' },
    { value: 'completed', label: 'Completed' },
    { value: 'available', label: 'Available' },
  ];

  return (
    <div className="space-y-6">
      <div>
        <Label htmlFor="roomInformation">Room Information</Label>
        <textarea
          id="roomInformation"
          className="w-full p-3 border border-gray-300 rounded-md focus:ring-2 focus:ring-blue-500 focus:border-transparent"
          rows={6}
          value={room.roomInformation || ''}
          onChange={(e) => handleInputChange('roomInformation', e.target.value)}
          placeholder="Enter detailed room information, policies, and any special notes..."
        />
      </div>

      <div>
        <Label htmlFor="status">Room Status</Label>
        <Select value={room.status || 'available'} onValueChange={(value) => handleInputChange('status', value)}>
          <SelectTrigger id="status">
            <SelectValue placeholder="Select room status" />
          </SelectTrigger>
          <SelectContent>
            {statusOptions.map((option) => (
              <SelectItem key={option.value} value={option.value}>
                {option.label}
              </SelectItem>
            ))}
          </SelectContent>
        </Select>
      </div>
    </div>
  );
};
