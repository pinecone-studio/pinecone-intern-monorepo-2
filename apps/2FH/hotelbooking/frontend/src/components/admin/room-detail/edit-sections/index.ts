export { BasicInfoSection } from './BasicInfoSection';
export { AmenitiesSection } from './AmenitiesSection';
export { ImagesSection } from './ImagesSection';
export { DetailsSection } from './DetailsSection';


