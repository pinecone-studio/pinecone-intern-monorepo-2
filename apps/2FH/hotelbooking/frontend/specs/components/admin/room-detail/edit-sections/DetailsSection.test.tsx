/* eslint-disable  */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { DetailsSection } from '@/components/admin/room-detail/edit-sections/DetailsSection';

// Mock the Select component to avoid Radix UI issues
jest.mock('@/components/ui/select', () => ({
  Select: ({ children, value, onValueChange }: any) => (
    <div data-testid="select">
      <button data-testid="select-trigger" id="status" onClick={() => onValueChange && onValueChange('booked')}>
        {value || 'Select room status'}
      </button>
      {children}
    </div>
  ),
  SelectContent: ({ children }: any) => <div data-testid="select-content">{children}</div>,
  SelectItem: ({ children, value }: any) => <div data-testid={`select-item-${value}`}>{children}</div>,
  SelectTrigger: ({ children }: any) => <div data-testid="select-trigger-wrapper">{children}</div>,
  SelectValue: ({ placeholder }: any) => <span data-testid="select-value">{placeholder}</span>,
}));

// Mock the Label component
jest.mock('@/components/ui/label', () => ({
  Label: ({ children, htmlFor }: any) => (
    <label htmlFor={htmlFor} data-testid="label">
      {children}
    </label>
  ),
}));

describe('DetailsSection', () => {
  const mockRoom = {
    roomInformation: 'This is a detailed room description with amenities and policies.',
    status: 'available',
  };

  const defaultProps = {
    room: mockRoom,
    handleInputChange: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Rendering', () => {
    it('renders all form fields correctly', () => {
      render(<DetailsSection {...defaultProps} />);

      expect(screen.getByLabelText('Room Information')).toBeInTheDocument();
      expect(screen.getByText('Room Status')).toBeInTheDocument();
      expect(screen.getByTestId('select')).toBeInTheDocument();
    });

    it('displays current room values', () => {
      render(<DetailsSection {...defaultProps} />);

      expect(screen.getByDisplayValue(mockRoom.roomInformation)).toBeInTheDocument();
      expect(screen.getByTestId('select-trigger')).toHaveTextContent('available');
    });

    it('renders with correct structure and classes', () => {
      render(<DetailsSection {...defaultProps} />);

      const container = screen.getByLabelText('Room Information').closest('div')?.parentElement;
      expect(container).toHaveClass('space-y-6');
    });

    it('renders textarea with correct attributes', () => {
      render(<DetailsSection {...defaultProps} />);

      const textarea = screen.getByLabelText('Room Information');
      expect(textarea).toHaveAttribute('id', 'roomInformation');
      expect(textarea).toHaveAttribute('rows', '6');
      expect(textarea).toHaveClass('w-full', 'p-3', 'border', 'border-gray-300', 'rounded-md');
    });

    it('renders select with correct attributes', () => {
      render(<DetailsSection {...defaultProps} />);

      const select = screen.getByTestId('select');
      expect(select).toBeInTheDocument();
      expect(screen.getByTestId('select-trigger')).toHaveAttribute('id', 'status');
    });
  });

  describe('User Interactions', () => {
    it('calls handleInputChange when room information changes', async () => {
      const user = userEvent.setup();
      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      await user.clear(roomInfoTextarea);
      await user.type(roomInfoTextarea, 'Updated room information');

      // Check that handleInputChange was called with the correct field name
      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', expect.any(String));
      // Check that it was called multiple times (once for each character)
      expect(defaultProps.handleInputChange).toHaveBeenCalledTimes(25);
    });

    it('calls handleInputChange when status changes via Select onValueChange', () => {
      render(<DetailsSection {...defaultProps} />);

      const selectTrigger = screen.getByTestId('select-trigger');
      fireEvent.click(selectTrigger);

      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('status', 'booked');
    });

    it('handles multiple rapid changes to room information', async () => {
      const user = userEvent.setup();
      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');

      await user.clear(roomInfoTextarea);
      await user.type(roomInfoTextarea, 'First change');
      await user.clear(roomInfoTextarea);
      await user.type(roomInfoTextarea, 'Second change');

      // userEvent.type triggers onChange for each character, so we check that it was called multiple times
      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', expect.any(String));
      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', '');
      // Should be called many times due to character-by-character typing
      expect(defaultProps.handleInputChange).toHaveBeenCalledTimes(27);
    });
  });

  describe('Edge Cases and Null Handling', () => {
    it('handles room with null/undefined values', () => {
      const roomWithNulls = { roomInformation: null, status: undefined };
      render(<DetailsSection {...defaultProps} room={roomWithNulls} />);

      expect(screen.getByLabelText('Room Information')).toHaveValue('');
      expect(screen.getByTestId('select-trigger')).toHaveTextContent('available');
    });

    it('handles room with empty string values', () => {
      const roomWithEmpty = { roomInformation: '', status: '' };
      render(<DetailsSection {...defaultProps} room={roomWithEmpty} />);

      expect(screen.getByLabelText('Room Information')).toHaveValue('');
      expect(screen.getByTestId('select-trigger')).toHaveTextContent('available');
    });

    it('handles room with missing properties', () => {
      const roomWithMissing = {};
      render(<DetailsSection {...defaultProps} room={roomWithMissing} />);

      expect(screen.getByLabelText('Room Information')).toHaveValue('');
      expect(screen.getByTestId('select-trigger')).toHaveTextContent('available');
    });

    it('handles room with null status', () => {
      const roomWithNullStatus = { ...mockRoom, status: null };
      render(<DetailsSection {...defaultProps} room={roomWithNullStatus} />);

      expect(screen.getByTestId('select-trigger')).toHaveTextContent('available');
    });

    it('handles room with undefined status', () => {
      const roomWithUndefinedStatus = { ...mockRoom, status: undefined };
      render(<DetailsSection {...defaultProps} room={roomWithUndefinedStatus} />);

      expect(screen.getByTestId('select-trigger')).toHaveTextContent('available');
    });
  });

  describe('Status Options', () => {
    it('handles different status values correctly', () => {
      const statuses = ['booked', 'cancelled', 'pending', 'completed', 'available'] as const;

      statuses.forEach((status) => {
        const { rerender } = render(<DetailsSection {...defaultProps} room={{ ...mockRoom, status }} />);
        expect(screen.getByTestId('select-trigger')).toHaveTextContent(status);
        rerender(<div />);
      });
    });

    it('handles all status options', () => {
      const statusOptions = [
        { value: 'cancelled', label: 'Cancelled' },
        { value: 'booked', label: 'Booked' },
        { value: 'pending', label: 'Pending' },
        { value: 'completed', label: 'Completed' },
        { value: 'available', label: 'Available' },
      ];

      statusOptions.forEach((option) => {
        const { rerender } = render(<DetailsSection {...defaultProps} room={{ ...mockRoom, status: option.value }} />);
        expect(screen.getByTestId('select-trigger')).toHaveTextContent(option.value);
        rerender(<div />);
      });
    });

    it('handles status change with different values', () => {
      const { rerender } = render(<DetailsSection {...defaultProps} />);

      const statuses = ['cancelled', 'booked', 'pending', 'completed'];

      statuses.forEach((status) => {
        rerender(<DetailsSection {...defaultProps} room={{ ...mockRoom, status }} />);
        expect(screen.getByTestId('select-trigger')).toHaveTextContent(status);
      });
    });
  });

  describe('Text Content and Special Characters', () => {
    it('handles roomInformation with special characters', () => {
      const roomWithSpecialChars = {
        ...mockRoom,
        roomInformation: 'Room with special chars: @#$%^&*()_+-=[]{}|;:,.<>?',
      };

      render(<DetailsSection {...defaultProps} room={roomWithSpecialChars} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      fireEvent.change(roomInfoTextarea, { target: { value: 'New content with special chars: !@#$%' } });

      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', 'New content with special chars: !@#$%');
    });

    it('handles very long room information text', () => {
      const longText = 'A'.repeat(1000);

      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      fireEvent.change(roomInfoTextarea, { target: { value: longText } });

      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', longText);
    });

    it('handles roomInformation with newlines', () => {
      const multilineText = 'Line 1\nLine 2\nLine 3';

      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      fireEvent.change(roomInfoTextarea, { target: { value: multilineText } });

      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', multilineText);
    });

    it('handles roomInformation with HTML-like content', () => {
      const htmlLikeText = '<div>Room info</div><script>alert("test")</script>';

      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      fireEvent.change(roomInfoTextarea, { target: { value: htmlLikeText } });

      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', htmlLikeText);
    });

    it('handles empty roomInformation after clearing', () => {
      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      fireEvent.change(roomInfoTextarea, { target: { value: '' } });

      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('roomInformation', '');
    });
  });

  describe('Placeholder and Labels', () => {
    it('displays correct placeholder text for room information', () => {
      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByPlaceholderText('Enter detailed room information, policies, and any special notes...');
      expect(roomInfoTextarea).toBeInTheDocument();
    });

    it('displays correct labels', () => {
      render(<DetailsSection {...defaultProps} />);

      expect(screen.getByText('Room Information')).toBeInTheDocument();
      expect(screen.getByText('Room Status')).toBeInTheDocument();
    });
  });

  describe('Component Props', () => {
    it('handles handleInputChange being undefined', () => {
      const propsWithoutHandler = { ...defaultProps, handleInputChange: undefined };

      // Should not throw error
      expect(() => render(<DetailsSection {...propsWithoutHandler} />)).not.toThrow();
    });

    it('handles room prop being undefined', () => {
      const propsWithoutRoom = { ...defaultProps, room: undefined };

      // Should throw error since component doesn't handle undefined room
      expect(() => render(<DetailsSection {...propsWithoutRoom} />)).toThrow();
    });
  });

  describe('Accessibility', () => {
    it('has proper label associations', () => {
      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      expect(roomInfoTextarea).toHaveAttribute('id', 'roomInformation');

      const statusSelect = screen.getByTestId('select-trigger');
      expect(statusSelect).toHaveAttribute('id', 'status');
    });

    it('has proper focus management', () => {
      render(<DetailsSection {...defaultProps} />);

      const roomInfoTextarea = screen.getByLabelText('Room Information');
      roomInfoTextarea.focus();
      expect(roomInfoTextarea).toHaveFocus();
    });
  });

  describe('Snapshot Tests', () => {
    it('matches snapshot with default props', () => {
      const { container } = render(<DetailsSection {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with null values', () => {
      const roomWithNulls = { roomInformation: null, status: undefined };
      const { container } = render(<DetailsSection {...defaultProps} room={roomWithNulls} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with different status', () => {
      const roomWithStatus = { ...mockRoom, status: 'booked' };
      const { container } = render(<DetailsSection {...defaultProps} room={roomWithStatus} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
