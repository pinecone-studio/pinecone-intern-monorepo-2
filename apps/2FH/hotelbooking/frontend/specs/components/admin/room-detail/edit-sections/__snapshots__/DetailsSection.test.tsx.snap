// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsSection Snapshot Tests matches snapshot with default props 1`] = `
<div
  class="space-y-6"
>
  <div>
    <label
      data-testid="label"
      for="roomInformation"
    >
      Room Information
    </label>
    <textarea
      class="w-full p-3 border border-gray-300 rounded-md focus:ring-2 focus:ring-blue-500 focus:border-transparent"
      id="roomInformation"
      placeholder="Enter detailed room information, policies, and any special notes..."
      rows="6"
    >
      This is a detailed room description with amenities and policies.
    </textarea>
  </div>
  <div>
    <label
      data-testid="label"
      for="status"
    >
      Room Status
    </label>
    <div
      data-testid="select"
    >
      <button
        data-testid="select-trigger"
        id="status"
      >
        available
      </button>
      <div
        data-testid="select-trigger-wrapper"
      >
        <span
          data-testid="select-value"
        >
          Select room status
        </span>
      </div>
      <div
        data-testid="select-content"
      >
        <div
          data-testid="select-item-cancelled"
        >
          Cancelled
        </div>
        <div
          data-testid="select-item-booked"
        >
          Booked
        </div>
        <div
          data-testid="select-item-pending"
        >
          Pending
        </div>
        <div
          data-testid="select-item-completed"
        >
          Completed
        </div>
        <div
          data-testid="select-item-available"
        >
          Available
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DetailsSection Snapshot Tests matches snapshot with different status 1`] = `
<div
  class="space-y-6"
>
  <div>
    <label
      data-testid="label"
      for="roomInformation"
    >
      Room Information
    </label>
    <textarea
      class="w-full p-3 border border-gray-300 rounded-md focus:ring-2 focus:ring-blue-500 focus:border-transparent"
      id="roomInformation"
      placeholder="Enter detailed room information, policies, and any special notes..."
      rows="6"
    >
      This is a detailed room description with amenities and policies.
    </textarea>
  </div>
  <div>
    <label
      data-testid="label"
      for="status"
    >
      Room Status
    </label>
    <div
      data-testid="select"
    >
      <button
        data-testid="select-trigger"
        id="status"
      >
        booked
      </button>
      <div
        data-testid="select-trigger-wrapper"
      >
        <span
          data-testid="select-value"
        >
          Select room status
        </span>
      </div>
      <div
        data-testid="select-content"
      >
        <div
          data-testid="select-item-cancelled"
        >
          Cancelled
        </div>
        <div
          data-testid="select-item-booked"
        >
          Booked
        </div>
        <div
          data-testid="select-item-pending"
        >
          Pending
        </div>
        <div
          data-testid="select-item-completed"
        >
          Completed
        </div>
        <div
          data-testid="select-item-available"
        >
          Available
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DetailsSection Snapshot Tests matches snapshot with null values 1`] = `
<div
  class="space-y-6"
>
  <div>
    <label
      data-testid="label"
      for="roomInformation"
    >
      Room Information
    </label>
    <textarea
      class="w-full p-3 border border-gray-300 rounded-md focus:ring-2 focus:ring-blue-500 focus:border-transparent"
      id="roomInformation"
      placeholder="Enter detailed room information, policies, and any special notes..."
      rows="6"
    />
  </div>
  <div>
    <label
      data-testid="label"
      for="status"
    >
      Room Status
    </label>
    <div
      data-testid="select"
    >
      <button
        data-testid="select-trigger"
        id="status"
      >
        available
      </button>
      <div
        data-testid="select-trigger-wrapper"
      >
        <span
          data-testid="select-value"
        >
          Select room status
        </span>
      </div>
      <div
        data-testid="select-content"
      >
        <div
          data-testid="select-item-cancelled"
        >
          Cancelled
        </div>
        <div
          data-testid="select-item-booked"
        >
          Booked
        </div>
        <div
          data-testid="select-item-pending"
        >
          Pending
        </div>
        <div
          data-testid="select-item-completed"
        >
          Completed
        </div>
        <div
          data-testid="select-item-available"
        >
          Available
        </div>
      </div>
    </div>
  </div>
</div>
`;
